catalogue/webapp/components/Calendar/Calendar.tsx

```diff
@@ -0,0 +1,278 @@
+import { FC, useState, useEffect, useRef } from 'react';
+import { Moment } from 'moment';
+import { DayNumber } from '@weco/common/model/opening-hours';
+import { classNames, font } from '@weco/common/utils/classnames';
+import {
+  getCalendarRows,
+  firstDayOfWeek,
+  lastDayOfWeek,
+} from './calendar-utils';
+import { isRequestableDate } from '../../utils/dates';
+import { DatePicker, Header, Table, Td } from './CalendarStyles';
+
+const LEFT = [37, 'ArrowLeft'];
+const RIGHT = [39, 'ArrowRight'];
+const DOWN = [40, 'ArrowDown'];
+const UP = [38, 'ArrowUp'];
+const HOME = [36, 'Home'];
+const END = [35, 'End'];
+const PAGEUP = [33, 'PageUp'];
+const PAGEDOWN = [34, 'PageDown'];
+const ENTER = [13, 'Enter'];
+
+function newDate(date: Moment, key: number | string): Moment {
+  const dates = getCalendarRows(date);
+  switch (true) {
```

Comment:
Nit/subjective opinion: I think I would take advantage of the switch so we don't have to invoke the arrays here, and also so it's a typical switch/case statement:
```typescript
switch (key) {
  // LEFT
  case 'ArrowLeft':
  case 37:
    return date.clone().subtract(1, 'day');
  // etc...
}
```
also, do we need `break`s if we have `return`s? I can never remember the rules with that

Reply:
Having the keys in an array is useful [here](https://github.com/wellcomecollection/wellcomecollection.org/blob/7cefcfc29b0ae3ba4710df3d1596ceca701f3f91/catalogue/webapp/components/Calendar/Calendar.tsx#L82) for seeing if we need to handle the event ourselves. Otherwise I'd have to keep them in 2 places.